sys/include/tsrb.h

```diff
@@ -46,15 +47,17 @@ typedef struct tsrb {
 /**
  * @brief Static initializer
  */
-#define TSRB_INIT(BUF) { (BUF), sizeof (BUF), 0, 0 }
+/* XXX remove this implicit cast as soon as possible (requires API change
+ *     to isrpipe)*/
+#define TSRB_INIT(BUF) { (uint8_t *)(BUF), sizeof (BUF), 0, 0 }
 
 /**
  * @brief        Initialize a tsrb.
  * @param[out]   rb        Datum to initialize.
  * @param[in]    buffer    Buffer to use by tsrb.
  * @param[in]    bufsize   `sizeof (buffer)`, must be power of 2.
  */
-static inline void tsrb_init(tsrb_t *rb, char *buffer, unsigned bufsize)
+static inline void tsrb_init(tsrb_t *rb, void *buffer, unsigned bufsize)
```

Comment:
why ```void *``` here? Is this "passing around"? IMO ```uint8_t *``` is nicer, especially for consistency.

Reply:
Porting artefact. I originally intended to make all buffers `void *`, but that would have required some additional casting to allow for the pointer arithmetics in some of the functions. Will fix.

Reply:
Then again... if I do this, all upper-layer users need to be adapted for this. Should I still do it?

Reply:
Done but not squashed yet. You decide :-)

Reply:
Yes please. Users of tsrb will have to touch their code anyways with this API change, so let's go uint8_t all the way. Easier to say "all buffer parameters are now uint8_t".

Reply:
Kk. Shall I squash then?

Reply:
ok!